routes/api_camps_routes.js

```diff
@@ -165,18 +165,26 @@ var __camps_update_status = (current_event_id, camp_id, user_id, action, camp_mg
    
                 };
 
-                //select the addinfo_json column from the camp member table
-                knex(constants.CAMP_MEMBERS_TABLE_NAME).select('user_id','addinfo_json')
+                // select the addinfo_json column from the camp member table
+                knex(constants.CAMP_MEMBERS_TABLE_NAME)
+                .select('user_id',`${constants.CAMP_MEMBERS_TABLE_NAME}.addinfo_json`,`${constants.EVENTS_TABLE_NAME}.addinfo_json as eventInfo`)
+                .rightJoin(constants.EVENTS_TABLE_NAME,`${constants.EVENTS_TABLE_NAME}.event_id`,`${constants.EVENTS_TABLE_NAME}.event_id`)
                 .where({
                     camp_id : userData.camp_id, 
-                    user_id : userData.user_id
+                    user_id : userData.user_id,
+                    event_id: current_event_id
                 })
                 .then(resp => {
+                    // checking that update of the pre sale ticket allocation is inside the valid time period
+                    const start = new Date(JSON.parse(resp[0].eventInfo).start_presale_tickets_allocation);
+                    const end = new Date(JSON.parse(resp[0].eventInfo).end_presale_tickets_allocation);
+                    const now = new Date();
+                    const isValidAllocationDate = start < now && now < end;
```

Comment:
@guyzohar1 - this is the server side presale allocation dates control.
lines 167-175 :I've added the current event => events.addinfo_json column to the knex query.
lines 178-182: I then use to extract the dates, and confirm it is a valid date to update.